### PR TITLE
Mary Lang Completed Iteration 4

### PR DESCRIPTION
**Pull request by mcat56**

### CODE REVIEW

File: test/vendor_test.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+  def test_stock
+    @vendor.stock("Peaches", 30)
+
+    assert_equal 30, @vendor.check_stock("Peaches")
+  end
+
+
+
+end
```
Comment (iandouglas) [inline test/vendor_test.rb]: great job here

File: lib/vendor.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+  def check_stock(item)
+    @inventory[item]
+  end
+
+  def stock(item, count)
+    @inventory[item] += count
+  end
+
+end
```
Comment (iandouglas) [inline lib/vendor.rb]: excellent job here

File: test/market_test.rb [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  def test_sell
+    @market.add_vendor(@vendor_1)
+    @market.add_vendor(@vendor_2)
+    @market.add_vendor(@vendor_3)
+
+
+    assert_equal false, @market.sell("Onions", 1)
+    assert_equal true, @market.sell("Banana Nice Cream", 5)
+    assert_equal true, @market.sell("Peaches", 40)
+  end
```
Comment (iandouglas) [inline test/market_test.rb]: it'd be good to check the case where you DID have a product at the market, but not enough in stock from all vendors, like `.sell("Peaches", 1000)` should have returned false as well

File: lib/market.rb [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+  end
+
+  def vendors_that_sell(item)
+    vendors_that_sell = []
+    @vendors.each do |vendor|
+      if vendor.inventory[item] > 0
+        vendors_that_sell << vendor
+      end
+    end
+    vendors_that_sell
```
Comment (iandouglas) [inline lib/market.rb]: this is good, there's an enumerator you can use that can get this data in less code

File: lib/market.rb [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+  end
+
+  def sorted_item_list
+    items = []
+    @vendors.each do |vendor|
+      items << vendor.inventory.keys
+    end
+    items.flatten.uniq.sort
```
Comment (iandouglas) [inline lib/market.rb]: excellent job uniq'ing before sorting

File: lib/market.rb [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+  end
+
+  def total_inventory
+    @vendors.each do |vendor|
+      vendor.inventory.each do |item,count|
+        @total[item] += count
+      end
+    end
+    @total
```
Comment (iandouglas) [inline lib/market.rb]: good job here

File: lib/market.rb [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+        elsif vendor.inventory[item] > 0 && count < vendor.inventory[item]
+          vendor.inventory[item] -= count
+          counts -= count
+        elsif vendor.inventory[item] > 0 && count > vendor.inventory[item]
+          vendor.inventory[item] = 0
+          counts -= count
+        end
+      end
+    end
+    true
```
Comment (iandouglas) [inline lib/market.rb]: good job here, can you think of any ways to DRY this up a little?

File: lib/market.rb (hunk 4 above)
Comment (mcat56) [reply lib/market.rb]: def vendors_that_sell(item)
    vendors_that_sell = @vendors.select {|vendor| vendor.inventory.include?(item)}
  end

File: test/market_test.rb (hunk 3 above)
Comment (mcat56) [reply test/market_test.rb]: if @total[item] == 0 || @total[item] < count
      return false
    end
I think this was there?

File: lib/market.rb (hunk 7 above)
Comment (mcat56) [reply lib/market.rb]: I really struggle drying up when I have if's

**Pull request closed without merging**